packages/notebook/src/windowing.ts

```diff
@@ -10,10 +10,107 @@ import {
   WindowedListModel
 } from '@jupyterlab/ui-components';
 import { Message, MessageLoop } from '@lumino/messaging';
-import { Debouncer } from '@lumino/polling';
+import { Debouncer, Throttler } from '@lumino/polling';
 import { Widget } from '@lumino/widgets';
 import { DROP_SOURCE_CLASS, DROP_TARGET_CLASS } from './constants';
 
+/**
+ * Check whether the element is in a scrolling notebook.
+ * Traverses open shadow DOM roots if needed.
+ */
+export function isInScrollingNotebook(element: Element | null): boolean {
+  if (!element) {
+    return false;
+  }
+  const notebook = element.closest('.jp-WindowedPanel-viewport') as
+    | HTMLElement
+    | undefined;
+  if (notebook && notebook.dataset.isScrolling == 'true') {
+    return true;
+  }
+  const root = element.getRootNode();
+  return !!(
+    root &&
+    root instanceof ShadowRoot &&
+    isInScrollingNotebook(root.host)
+  );
+}
+
+/**
+ * Subclass IntersectionObserver to allow suspending callbacks when notebook is scrolling.
+ */
+window.IntersectionObserver = class extends window.IntersectionObserver {
```

Comment:
Out of curiosity, where is this observer attached to the cell ? I can't find any use of `IntersectionObserver` in the codebase of Jupyterlab, except for the `logconsole`...

Reply:
It is not in the JupyterLab codebase, but outputs can contain aribtrary this party widgets. The issue #16326 highlights the case of panel tabulator which has both resize and intersection observer and these fire on scrolling causing resizing of the viewport.

Reply:
Thanks for the answer.
I was testing this PR with `jupyterlab_myst` and it seems that it fixes https://github.com/jupyter-book/jupyterlab-myst/issues/243, and I was thinking it was because of this `IntersectionObserver`, this is why I was confused.
I just realised that the change related to the `myst` issue is probably 
```ts
const requiresSoftHiding = widget.node.querySelector('defs,.myst');
```

Reply:
Yes, as documented by the comment:

```js
    // - elements containing `.myst` class as jupyterlab-myst
    //   re-renders cause height jitter and scrolling issues;
    //   in future this could be addressed by the proposal
    //   to allow renderers to specify hiding mode, see
    //   https://github.com/jupyterlab/jupyterlab/issues/17331
```

I do not like special-casing `myst` here, but given the prominence of that extension (and it being a part of an official Jupyter subproject) I feel this is warranted.